### ISSUE TITLE
Bug: forwardRef provides an empty object as ref, causing `can't define property "current"` error.

### ISSUE THREAD

**Issue by hansottowirtz** [facebook/react#26419]
React version: 18.3.0-next-3ba7add60-20221201

Hard to reproduce.

In Next.js, in rare cases, and only after a fast refresh, the ref object when using forwardRef is simply `{}`.

This leads to the `TypeError: can't define property "current": Object is not extensible` error when used with `useImperativeHandle`.

```tsx
function Parent() {
  const carouselRef = useRef<any>();

  return <Suspense><Carousel ref={carouselRef} /></Suspense>
}

const Carousel = forwardRef((props, ref) => {
  if (!('current' in ref)) { debugger; }
  const thing = use(aPromise);
  return ...;
});
```

![Screenshot 2023-03-17 at 18 49 21](https://user-images.githubusercontent.com/5406212/225981809-22f575fa-0ec7-4f94-ae23-bffc8bf7c4dd.png)

In `updateImperativeHandle`, the ref is already an empty object.

![Screenshot 2023-03-17 at 19 13 13](https://user-images.githubusercontent.com/5406212/225985622-b8512a61-3a6b-4958-b63f-277219d4c816.png)

I haven't investigated it further yet, the stacktraces are quite hard to follow.

**Comment by joaoromeira** [comment]
You should try this approach (I think you want to use the ref to control the carousel)

```jsx
import { useEffect } from 'react';
import { forwardRef, Suspense, useImperativeHandle, useRef } from 'react'


type CarouselProps = any

type CarouselHandle = {
  init: () => void
}

const Carousel  = forwardRef<CarouselHandle, CarouselProps>((props, ref) => {
  useImperativeHandle(ref, () => ({
   init: () => {
     alert('Init!')
   } 
  }))

  return <div>Carousel</div>
});


const Parent = () => {
  const carouselRef = useRef<CarouselHandle>(null);
  
  useEffect(() => {
    if(carouselRef.current)
      carouselRef.current.init()
  }, [carouselRef])
  return (
    <Suspense fallback={<div />}>
      <Carousel ref={carouselRef} />
    </Suspense>
  )
}

export default Parent
```

**Comment by eps1lon** [comment]
Can you [open this on the Next.js repo](https://github.com/vercel/next.js/issues/new/choose) as well? They might have a better idea how to reduce this to a repro just using React.

**Comment by hansottowirtz** [comment]
@eps1lon I have created a failing test/reproduction in #26420, I'm waiting for some help over there.

**Comment by hansottowirtz** [comment]
Fixed by #26535
